### PR TITLE
refactor: [M3-6287] - MUI v5 `Features > Users`

### PR DESCRIPTION
**Pull request by carrillo-erik**
## Description 📝
MUI v5 migration for `Users`

## Major Changes 🔄
- Ran `tss-react` codemod on `src > features > Users`.
- Replaced default export with named export.
- Removed `index.tsx` file.

## Preview 📷
The work on this PR is contained to only migrate the styling to the newer API.

## How to test 🧪
1. Verify the functionality around the `Users` feature works as expected.
2. Look for any UI defects or regressions in the `Users` landing page.
3. Visit `account/users` and validate adding new users and changing the `User Permissions`.

### CODE REVIEW

File: packages/manager/src/features/Users/CreateUserDrawer.tsx [hunk 1]
```diff
@@ -5,14 +5,14 @@ import { RouteComponentProps, withRouter } from 'react-router-dom';
...
+import { FormControlLabel } from 'src/components/FormControlLabel';
 import { Notice } from 'src/components/Notice/Notice';
 import { TextField } from 'src/components/TextField';
 import { Toggle } from 'src/components/Toggle';
-import { FormControlLabel } from 'src/components/FormControlLabel';
 import { getAPIErrorOrDefault } from 'src/utilities/errorUtils';
 import getAPIErrorsFor from 'src/utilities/getAPIErrorFor';
 
-interface Props {
+interface CreateUserDrawerProps {
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Users/CreateUserDrawer.tsx:15]: We decided that it's okay to keep `interface Props` if the props are not exported / shared throughout the app. This change is fine, but consider keeping it `Props` in the future.

File: packages/manager/src/features/Users/UserDeleteConfirmationDialog.tsx [hunk 2]
```diff
@@ -3,16 +3,19 @@ import * as React from 'react';
 import { ActionsPanel } from 'src/components/ActionsPanel/ActionsPanel';
 import { ConfirmationDialog } from 'src/components/ConfirmationDialog/ConfirmationDialog';
 
-interface Props {
+interface UserDeleteConfirmationDialogProps {
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Users/UserDeleteConfirmationDialog.tsx:6]: This change is fine, but consider keeping it `Props` in the future.

File: packages/manager/src/features/Users/UserPermissionsEntitySection.tsx [hunk 3]
```diff
@@ -26,70 +35,18 @@ export const entityNameMap: Record<GrantType, string> = {
   vpc: 'VPCs',
 };
 
-interface Props {
+interface UserPermissionsEntitySectionProps {
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Users/UserPermissionsEntitySection.tsx:38]: Same `Props` feedback here.

File: packages/manager/src/features/Users/UsersActionMenu.tsx [hunk 4]
```diff
@@ -8,19 +8,19 @@ import { Action, ActionMenu } from 'src/components/ActionMenu';
 import { InlineMenuAction } from 'src/components/InlineMenuAction/InlineMenuAction';
 import { useProfile } from 'src/queries/profile';
 
-interface Props {
+interface UsersActionMenuProps {
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Users/UsersActionMenu.tsx:11]: Same `Props` feedback

**Review by bnussman-akamai — APPROVED** [review]
Found no new defects on the Account users page 

It would be nice if we could fix the possitioning of these buttons that broke due to https://github.com/linode/manager/pull/9341 but this may be out of scope of this ticket 

Screenshot 2023-10-04 at 9 44 05 AM

**Review by abailly-akamai — APPROVED** [review]
Looks good! only found one small visual regression

Screenshot 2023-10-04 at 10 38 17

**Comment by carrillo-erik** [comment]
> Found no new defects on the Account users page
> 
> It would be nice if we could fix the positioning of these buttons that broke due to #9341 but this may be out of scope of this ticket
> 
> Screenshot 2023-10-04 at 9 44 05 AM


I went ahead and took care of fixing this UI regression within this PR. 

Screenshot 2023-10-05 at 4 43 05 AM

File: packages/manager/src/features/Users/CreateUserDrawer.tsx (hunk 1 above)
Comment (carrillo-erik) [reply packages/manager/src/features/Users/CreateUserDrawer.tsx]: Reverted to `interface Props`.

File: packages/manager/src/features/Users/UserDeleteConfirmationDialog.tsx (hunk 2 above)
Comment (carrillo-erik) [reply packages/manager/src/features/Users/UserDeleteConfirmationDialog.tsx]: Reverted to `interface Props`.

File: packages/manager/src/features/Users/UserPermissionsEntitySection.tsx (hunk 3 above)
Comment (carrillo-erik) [reply packages/manager/src/features/Users/UserPermissionsEntitySection.tsx]: Reverted to `interface Props`.

File: packages/manager/src/features/Users/UsersActionMenu.tsx (hunk 4 above)
Comment (carrillo-erik) [reply packages/manager/src/features/Users/UsersActionMenu.tsx]: Reverted to `interface Props`.

**Comment by carrillo-erik** [comment]
> Looks good! only found one small visual regression
> 
> Screenshot 2023-10-04 at 10 38 17

Nice catch! This has been fixed with my latest changes.

Screenshot 2023-10-05 at 5 06 07 AM

**Comment by abailly-akamai** [comment]
merging this on behalf of @carrillo-erik

**Pull request merged**